src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -46,17 +44,8 @@ class PeerDiscoveryManager(
   }
 
   def scan(): Unit = {
-    // At the beginning ping all bootstrap nodes
-    if (firstScan) {
-      bootStrapNodesInfo.foreach{ nodeInfo =>
-        sendPing(Endpoint.makeEndpoint(nodeInfo.node.udpSocketAddress, nodeInfo.node.tcpPort), nodeInfo.node.udpSocketAddress, nodeInfo)
-      }
-
-      firstScan = false
-    }
-
     // Ping a random sample from currently pinged nodes without the answer
-    new Random().shuffle(pingedNodes.values).take(discoveryConfig.scanMaxNodes).foreach {pingInfo =>
+    new Random().shuffle(pingedNodes.values).take(2 * discoveryConfig.scanMaxNodes).foreach {pingInfo =>
```

Comment:
Although we are pinging more nodes, it seems we are still storing bad nodes in nodesInfo. I did a hacky test to forward the PeerClosedConnection message to this actor and try to prune the nodes here, but didn't observe a great improvement. There probably would be an improvement but not sure it's worth the effort since this discovery implementation is temporary I believe (correct?) and the task was for the bug fix (which I tested and works), not for optimization.